doc/rst/language/spec/generics.rst

```diff
@@ -473,50 +473,77 @@ A queried domain may not be modified via the name to which it is bound
+and the point of instantiation is nested within generic function(s),
+the point of instantiation of the innermost enclosing generic function
+is consulted. This process is repeated until the point of instantiation
+for which candidate function(s) are found. If points of instantiation
+are exhausted without finding any candidate functions, the compiler
+issues a "call cannot be resolved" error.
 
    *Example (point-of-instantiation.chpl)*.
 
-   Consider the following code which defines a generic function ``bar``:
+   Consider the following code:
    
 
    .. code-block:: chapel
 
-      module M1 {
-        record R {
-          var x: int;
-          proc foo() { }
+      module LibraryA {
+        proc doitA(arg) {
+          worker1();
+          worker2();
         }
       }
 
-      module M2 {
-        proc bar(x) {
-          x.foo();
+      module LibraryB {
+        use LibraryA;
+        proc worker1() { writeln("in LibraryB"); }
+        proc doitB(arg) {
+          doitA(arg);
         }
       }
 
-      module M3 {
-        use M1, M2;
+      module Application {
+        use LibraryB;
+        proc worker1() { writeln("in Application"); }
+        proc worker2() { writeln("in Application"); }
         proc main() {
-          var r: R;
-          bar(r);
+          doitB(1);
         }
       }
 
-   In the function ``main``, the variable ``r`` is declared to be of
-   type ``R`` defined in module ``M1`` and a call is made to the generic
-   function ``bar`` which is defined in module ``M2``. This is the only
-   place where ``bar`` is called in this program and so it becomes the
-   point of instantiation for ``bar`` when the argument ``x`` is of type
-   ``R``. Therefore, the call to the ``foo`` method in ``bar`` is
-   resolved by looking for visible functions from within ``main`` and
-   going through the use of module ``M1``.
+   .. BLOCK-test-chapeloutput
+
+      in LibraryB
+      in Application
+
+   When resolving the calls to ``worker1`` and ``worker2`` in ``doitA()``
+   there are no visible functions at the scope of the call. Since
+   ``doitA()`` is a generic function, its point of instantiation is
+   therefore consulted, which is its call within ``doitB()``.
+   There, a single candidate function for ``worker1`` is found, so
+   function resolution determins that this is the target function.
+   No visible functions for ``worker2`` are found there, however,
+   so the point of instantiation of ``doitB()`` is consulted next.
+   It is the call to ``doitB`` in the ``Application`` module.
+   There a definition of ``worker2`` is visible and will be considered
+   the candidate for the call to ``worker2`` in ``doitA()``.
+
+   ``LibraryB`` relies on the callers of its ``doitB()`` to provide
+   implementations of ``worker2``. It is guaranteed that its own implementation
+   of ``worker1`` is used, regardless of overloads of ``worker1``,
+   if any, that are visible from ``doitB`` callers.
```

Comment:
It would be nice to emphasize that this property is a corollary of the previous rule. Could just be "Due to the order in which functions from points of instantiations are visited... " or something along those lines.